webapp/src/main/java/com/box/l10n/mojito/quartz/QuartzPollableTaskScheduler.java

```diff
@@ -161,6 +172,34 @@ public <I, O> PollableFuture<O> scheduleJob(QuartzJobInfo<I, O> quartzJobInfo) {
     return new QuartzPollableFutureTask<O>(pollableTask, jobOutputType);
   }
 
+  private <I, O> QuartzPollableFutureTask<O> getPendingPollableFutureIfExists(
+      QuartzJobInfo<I, O> quartzJobInfo, Scheduler scheduler) {
+    logger.debug(
+        "DisallowConcurrentExecution is set, checking if the job already has a trigger in a blocked state");
+    String jobKeyName = getKeyName(quartzJobInfo.getClazz(), quartzJobInfo.getUniqueId());
+    try {
+      List<Trigger> triggers =
+          (List<Trigger>) scheduler.getTriggersOfJob(new JobKey(jobKeyName, DYNAMIC_GROUP_NAME));
+      for (Trigger trigger : triggers) {
+        if (scheduler.getTriggerState(trigger.getKey()).equals(Trigger.TriggerState.BLOCKED)) {
+          // Should only ever be one trigger in this state when @DisallowConcurrentExecution is set
```

Comment:
The "blocked" trigger is for the running job right? Is there a job queued for the one that should run after?

Reply:
No, my understanding is that the running job trigger is in state COMPLETED as it has completed it 'triggering' and the job waiting to execute will be in the BLOCKED state, as it's blocked from executing until after the currently executing job finishes.

I'm running into an issue now with the unit test on GH, seems the RAMJobStore is returning BLOCKED for the executing job too. This doesn't lineup with the behaviour I've seen when running the tests against a mysql DB so trying to figure out what's going wrong there